examples/pngWithRoundedShapes.php

```diff
@@ -0,0 +1,210 @@
+
+        /**
+         * @var int $neighbours
+         * The right bit order (starting from 0):
+         *   0 1 2
+         *   7 # 3
+         *   6 5 4
+         */
+        $neighbours = $this->matrix->checkNeighbours($x, $y);
+
+        // ------------------
+        // Outer rounding
+        // ------------------
+
+        if ($neighbours & (1 << 7)) { // neighbour left
+            // top left
+            imagefilledrectangle(
+                $this->image,
+                $x1,
+                $y1,
+                $x1 + $rectsize,
+                $y1 + $rectsize,
+                $this->moduleValues[$M_TYPE]
+            );
+            // bottom left
+            imagefilledrectangle(
+                $this->image,
+                $x1,
+                $y2 - $rectsize,
+                $x1 + $rectsize,
+                $y2,
+                $this->moduleValues[$M_TYPE]
+            );
+        }
+
+        if ($neighbours & (1 << 3)) { // neighbour right
+            // top right
+            imagefilledrectangle(
+                $this->image,
+                $x2 - $rectsize,
+                $y1,
+                $x2,
+                $y1 + $rectsize,
+                $this->moduleValues[$M_TYPE]
+            );
+            // bottom right
+            imagefilledrectangle(
+                $this->image,
+                $x2 - $rectsize,
+                $y2 - $rectsize,
+                $x2,
+                $y2,
+                $this->moduleValues[$M_TYPE]
+            );
+        }
+
+        if ($neighbours & (1 << 1)) { // neighbour сверху
+            // top left
+            imagefilledrectangle(
+                $this->image,
+                $x1,
+                $y1,
+                $x1 + $rectsize,
+                $y1 + $rectsize,
+                $this->moduleValues[$M_TYPE]
+            );
+            // top right
+            imagefilledrectangle(
+                $this->image,
+                $x2 - $rectsize,
+                $y1,
+                $x2,
+                $y1 + $rectsize,
+                $this->moduleValues[$M_TYPE]
+            );
+        }
+
+        if ($neighbours & (1 << 5)) { // neighbour снизу
+            // bottom left
+            imagefilledrectangle(
+                $this->image,
+                $x1,
+                $y2 - $rectsize,
+                $x1 + $rectsize,
+                $y2,
+                $this->moduleValues[$M_TYPE]
+            );
+            // bottom right
+            imagefilledrectangle(
+                $this->image,
+                $x2 - $rectsize,
+                $y2 - $rectsize,
+                $x2,
+                $y2,
+                $this->moduleValues[$M_TYPE]
+            );
+        }
+
+        // ---------------------
+        // inner rounding
+        // ---------------------
+
+        if (!$this->matrix->check($x, $y)) {
+
+            if ($neighbours & 1 && $neighbours & (1 << 7) && $neighbours & (1 << 1)) {
```

Comment:
PHPCS will yell at this (and all other) "unbracketed operation"

The many braces but clean way would be:
```php
if (($neighbours & 1) && ($neighbours & (1 << 7)) && ($neighbours & (1 << 1))){}
```